### ISSUE TITLE
[4.1][Bug] Save and back returns to http_referrer even when the referrer is the create/update form

### ISSUE THREAD

**Issue by tabacitu** [Laravel-Backpack/CRUD#2328]
# Bug report

### What I did

I've just received a bug report by email, saying: 

My teammate discovered some weird bug on save and back method. Here is the thing:

- You create a new record, and click "save and edit". Works ok, will go back to edit form.
- Now on the same edit form, you click on "save and back". 
- Bug, is back to create form, and that is no working ok, we are waiting for this back to edit form.

To fix this, we need to edit "performSaveAction" method:

```php
case 'save_and_back':
            default:
                if (\Request::has('http_referrer') && strpos(\Request::get('http_referrer'), 'create') === false && strpos(\Request::get('http_referrer'), 'edit') === false) {
                    $redirectUrl = \Request::get('http_referrer');
                } else {
                    $redirectUrl = $this->route;
                }
                break;
```

### What I expected to happen

Return to edit form, not create.

### What happened

Looks like on Create, clicking Save and Edit:
- triggers the new bubble that says the default save action has been changed
- does not ACTUALLY remember to do "save and edit", it still does "save_and_back"

**Comment by tabacitu** [comment]
My problem with the solution above is that it checks if the "create" or "update" strings are present the in the referrer_url. But. The referrer_url might NOT be a create/update form, and still have those strings.

I think we need to find a better solution. And I'm sure there is one, but I've spend a reasonable amount on this already, and can't put my finger on it 😄 Maybe anybody else has an idea?

**Comment by pxpm** [comment]
Hey @tabacitu 
I am not sure if it was the same person, but i answered someone with same question.

I think the button is doing what's expected, because `save and back` it will save and go to previous page user was. Nothing new here. 

I think the only afordable solution would be to allow devs to override the save actions, otherwise add an option like `Save and List` that always go to entity listing page. Because `Save and Back` is intuitive and working as expected like i said, what those devs are looking are for the `Save and List` that always go back to list page. 

It's not a bug IMO. 

I will have a look in the options we have for the save options.

Best,
Pedro

**Comment by pxpm** [comment]
Getting fixed in #2356
